shell/models/provisioning.cattle.io.cluster.js

```diff
@@ -954,6 +971,34 @@ export default class ProvCluster extends SteveModel {
     if ( res?._status === 204 ) {
       await this.$dispatch('ws.resource.remove', { data: this });
     }
+
+    // If this cluster has a custom provisioner, allow it to do custom deletion
+    if (this.customProvisionerHelper?.postDelete) {
+      return this.customProvisionerHelper?.postDelete(this);
+    }
+  }
+
+  /**
+   * Get the custom provisioner helper for this model
+   */
+  get customProvisionerHelper() {
+    // Find the first model extension that says it can be used for this model
+    return this.modelExtensions.find((modelExt) => modelExt.useFor ? modelExt.useFor(this) : false);
```

Comment:
should there be a type / interface for modelExtension instances? missing this fn will trip up developers

Reply:
This code is javascript. The useFor is on the interface for the cluster provisioning extension - this was incorrectly `useForModel` - that has been fixed.